OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -0,0 +1,341 @@
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, string.Format("User \"{0}\" is forced in variation \"vtag1\".", WhitelistedUserId)), Times.Once);
+            // no attributes provided for a experiment that has an audience
+            Assert.IsTrue(TestData.CompareObjects(actualVariation, expectedVariation));
+            
+            //verify(decisionService).getWhitelistedVariation(experiment, whitelistedUserId);
+            //verify(decisionService, never()).getStoredVariation(eq(experiment), any(UserProfile.class));
+        }
+
+        [Test]
+        public void GetVariationEvaluatesUserProfileBeforeAudienceTargeting()
+        {
+            Experiment experiment = ValidProjectConfig.Experiments[0];
+            Variation variation = experiment.Variations[0];
+
+            Bucketer bucketer = new Bucketer(LoggerMock.Object);
+            Decision decision = new Decision(variation.Id);
+            UserProfile userProfile = new UserProfile(UserProfileId, new Dictionary<string, Decision>
+            {
+                { experiment.Id, decision }
+            });
+
+            UserProfileServiceMock.Setup(up => up.Lookup(WhitelistedUserId)).Returns(userProfile.ToMap());
+
+            DecisionService decisionService = new DecisionService(bucketer, ErrorHandlerMock.Object, ValidProjectConfig, UserProfileServiceMock.Object, LoggerMock.Object);
+
+            decisionService.GetVariation(experiment, GenericUserId, new UserAttributes());
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, string.Format("User \"{0}\" does not meet conditions to be in experiment \"{1}\".",
+                GenericUserId, experiment.Key)), Times.Once);
+
+            // ensure that a user with a saved user profile, sees the same variation regardless of audience evaluation
+            decisionService.GetVariation(experiment, UserProfileId, new UserAttributes());
+        }
+
+        [Test]
+        public void GetForcedVariationReturnsForcedVariation()
+        {
+            Bucketer bucketer = new Bucketer(LoggerMock.Object);
+            DecisionService decisionService = new DecisionService(bucketer, ErrorHandlerMock.Object, ValidProjectConfig, null, LoggerMock.Object);
+
+            Assert.IsTrue(TestData.CompareObjects(WhitelistedVariation, decisionService.GetWhitelistedVariation(WhitelistedExperiment, WhitelistedUserId)));
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, string.Format("User \"{0}\" is forced in variation \"{1}\".",
+                WhitelistedUserId, WhitelistedVariation.Key)), Times.Once);
+
+            //logbackVerifier.expectMessage(Level.INFO, "User \"" + whitelistedUserId + "\" is forced in variation \""
+            //        + whitelistedVariation.getKey() + "\".");
+
+            //assertEquals(whitelistedVariation, decisionService.getWhitelistedVariation(whitelistedExperiment, whitelistedUserId));
```

Comment:
Done. the commented code was already implemented, I just removed commented code.